### PR TITLE
D: update to work with newest dub(package manager)

### PR DESCRIPTION
**Pull request by rsw0x**
Dub(the D package manager) had a series of changes over the past few months, just adding a few fixes to keep it up to date/make use of the IDE-helper functions it added for spitting out dependencies. And it recognizes a few more compiler messages correctly.

I think everything looks OK, but I'm not too well versed in vimscript.

### CODE REVIEW

File: autoload/neomake/makers/ft/d.vim [hunk 1]
```diff
@@ -14,52 +14,66 @@ function! neomake#makers#ft#d#EnabledMakers()
     return l:makers
 endfunction
 
+function! s:findDubRoot()
+    "Look upwards for a dub.json or dub.sdl to find the root
+    "I did it like this because it's the only cross platform way I know of
+    let l:tmp_file = findfile("dub.json", ".;")
+    if(empty(l:tmp_file))
```
Comment (benekastah) [inline autoload/neomake/makers/ft/d.vim]: No need for parens around the condition

File: autoload/neomake/makers/ft/d.vim [hunk 2]
```diff
@@ -14,52 +14,66 @@ function! neomake#makers#ft#d#EnabledMakers()
...
+        let l:output = system(l:dubCmd . tmp_dir)
+        if(v:shell_error == 0 && !empty(l:output))
+            " Is \n portable?
+            let s:dubImports = split(l:output, '\n')
+            call map(s:dubImports, '"-I" . v:val')
+            return
+        endif
     endif
-    return l:dub_incs
+    let s:dubImports = []
```
Comment (benekastah) [inline autoload/neomake/makers/ft/d.vim]: This reads a bit backwards. I think this should go at the top of the function, and just set it again if dub is found. Then you also don't need to return early above.

**Comment by benekastah** [comment]
Thanks for this! Once you address my comments I'll merge :)

**Comment by rsw0x** [comment]
Sorry for the delay, was busy. Fixed the minor issues I believe.

**Comment by msoucy** [comment]
This looks very nice - any chance of getting it merged in soon?

**Comment by benekastah** [comment]
There you go!

**Pull request merged**